### ISSUE TITLE
CT item not found in array

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/fractions-common#112]
Happening in multiple fractions sims:
```
fractions-intro : multitouch-fuzz : unbuilt
https://bayes.colorado.edu/continuous-testing/ct-snapshots/1612448066994/fractions-intro/fractions-intro_en.html?continuousTest=%7B%22test%22%3A%5B%22fractions-intro%22%2C%22multitouch-fuzz%22%2C%22unbuilt%22%5D%2C%22snapshotName%22%3A%22snapshot-1612448066994%22%2C%22timestamp%22%3A1612449403099%7D&brand=phet&ea&fuzz&fuzzPointers=2&memoryLimit=1000
Query: brand=phet&ea&fuzz&fuzzPointers=2&memoryLimit=1000
Uncaught Error: Assertion failed: item not found in Array
Error: Assertion failed: item not found in Array
at window.assertions.assertFunction (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1612448066994/assert/js/assert.js:25:13)
at arrayRemove (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1612448066994/phet-core/js/arrayRemove.js:19:13)
at Proxy.remove (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1612448066994/axon/js/createObservableArray.js:345:33)
at IntroModel.completePiece (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1612448066994/fractions-common/js/intro/model/ContainerSetModel.js:160:17)
at IntroModel.emptyNextCell (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1612448066994/fractions-common/js/intro/model/ContainerSetModel.js:227:16)
at https://bayes.colorado.edu/continuous-testing/ct-snapshots/1612448066994/fractions-common/js/intro/model/ContainerSetModel.js:324:14
at baseTimes (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1612448066994/sherpa/lib/lodash-4.17.4.js:997:23)
at Function.times (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1612448066994/sherpa/lib/lodash-4.17.4.js:16095:20)
at IntroModel.onNumeratorChange (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1612448066994/fractions-common/js/intro/model/ContainerSetModel.js:319:7)
at TinyEmitter.emit (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1612448066994/axon/js/TinyEmitter.js:71:18)
id: Bayes Chrome
Snapshot from 2/4/2021, 7:14:26 AM
```

**Comment by KatieWoe** [comment]
```
fractions-intro : multitouch-fuzz : unbuilt
https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647595636252/fractions-intro/fractions-intro_en.html?continuousTest=%7B%22test%22%3A%5B%22fractions-intro%22%2C%22multitouch-fuzz%22%2C%22unbuilt%22%5D%2C%22snapshotName%22%3A%22snapshot-1647595636252%22%2C%22timestamp%22%3A1647606649706%7D&brand=phet&ea&fuzz&fuzzPointers=2&memoryLimit=1000&supportsPanAndZoom=false
Query: brand=phet&ea&fuzz&fuzzPointers=2&memoryLimit=1000&supportsPanAndZoom=false
Uncaught Error: Assertion failed: item not found in Array
Error: Assertion failed: item not found in Array
at window.assertions.assertFunction (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647595636252/assert/js/assert.js:25:13)
at arrayRemove (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647595636252/chipper/dist/js/phet-core/js/arrayRemove.js:19:13)
at Proxy.remove (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647595636252/chipper/dist/js/axon/js/createObservableArray.js:355:5)
at IntroModel.completePiece (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647595636252/chipper/dist/js/fractions-common/js/intro/model/ContainerSetModel.js:149:17)
at IntroModel.emptyNextCell (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647595636252/chipper/dist/js/fractions-common/js/intro/model/ContainerSetModel.js:221:16)
at https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647595636252/chipper/dist/js/fractions-common/js/intro/model/ContainerSetModel.js:312:14
at baseTimes (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647595636252/sherpa/lib/lodash-4.17.4.js:997:23)
at Function.times (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647595636252/sherpa/lib/lodash-4.17.4.js:16095:20)
at IntroModel.onNumeratorChange (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647595636252/chipper/dist/js/fractions-common/js/intro/model/ContainerSetModel.js:308:7)
at TinyProperty.emit (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647595636252/chipper/dist/js/axon/js/TinyEmitter.js:68:9)
id: Bayes Chrome
Snapshot from 3/18/2022, 3:27:16 AM
```

**Comment by KatieWoe** [comment]
May be connected:
```
build-a-fraction : pan-and-zoom-fuzz : unbuilt
https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647707554210/build-a-fraction/build-a-fraction_en.html?continuousTest=%7B%22test%22%3A%5B%22build-a-fraction%22%2C%22pan-and-zoom-fuzz%22%2C%22unbuilt%22%5D%2C%22snapshotName%22%3A%22snapshot-1647707554210%22%2C%22timestamp%22%3A1647799751705%7D&brand=phet&ea&fuzz&fuzzPointers=2&memoryLimit=1000&supportsPanAndZoom=true
Query: brand=phet&ea&fuzz&fuzzPointers=2&memoryLimit=1000&supportsPanAndZoom=true
Uncaught Error: Assertion failed: Trying to translate with a non-finite Vector2
Error: Assertion failed: Trying to translate with a non-finite Vector2
at window.assertions.assertFunction (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647707554210/assert/js/assert.js:25:13)
at AnimatedPanZoomListener.animateToTargets (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647707554210/chipper/dist/js/scenery/js/listeners/AnimatedPanZoomListener.js:898:19)
at AnimatedPanZoomListener.step (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647707554210/chipper/dist/js/scenery/js/listeners/AnimatedPanZoomListener.js:202:10)
at https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647707554210/chipper/dist/js/joist/js/Sim.js:267:43
at PhetioAction.execute (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647707554210/chipper/dist/js/tandem/js/PhetioAction.js:90:17)
at Sim.stepSimulation (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647707554210/chipper/dist/js/joist/js/Sim.js:822:31)
at Sim.stepOneFrame (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647707554210/chipper/dist/js/joist/js/Sim.js:811:12)
at Sim.runAnimationLoop (https://bayes.colorado.edu/continuous-testing/ct-snapshots/1647707554210/chipper/dist/js/joist/js/Sim.js:789:12)
id: Bayes Chrome
Snapshot from 3/19/2022, 10:32:34 AM
```

**Comment by marlitas** [comment]
This is happening in `fractions-equality`, `fractions-mixed-numbers` and `fractions-intro` as of 9/22/23

**Comment by jbphet** [comment]
Here is a sequence to reproduce the problem manually in Fractions Intro. This requires a touchscreen.

Load sim, select first screen
Increment the fraction denominator to one or more using the circular arrow buttons
Grab a piece from the bucket with one finger and hold it near the container in the center without releasing it
With another finger, press and release the Reset All button.
Lift the first finger. Note that the numerator of the fraction increases.
Decrease the numerator by one using the circular arrow buttons

**Comment by jbphet** [comment]
@marlitas and I looked at this with some consultation from @jonathanolson, and I have put it a fix that addresses the issue.  I'll keep an eye on CT, but @marlitas and I were a bit surprised that we couldn't solve the problem using `interruptSubtreeInput`, so we will talk a bit more with @jonathanolson and see if we can come up with a better solution.

**Comment by jbphet** [comment]
In https://github.com/phetsims/scenery/issues/1582 @jonathanolson committed some changes to joist and scenery that are intended to solve this problem in a more general way.  This is great, because it may help to resolve other CT issues that have been hard to track down.  I think the general changes are intended to be permanent, so the code I committed above can be reverted, but I want to double check with @jonathanolson before I undo them.

**Comment by jbphet** [comment]
@jonathanolson said to go ahead and remove the workaround, so I have done so.  I'll leave this open over the weekend and, if CT is clear of this problem on Monday, I'll close.

**Comment by jbphet** [comment]
Now that the workaround has been removed, the problem has reappeared on CT.  I did some testing, and below is the sequence to reproduce the problem.  The requires a touch screen.

1. Load Fractions Intro, select first screen
2. Grab a piece from the bucket with one finger and hold it near the container in the center without releasing it
3. With another finger, increment the fraction denominator by one.  This will cause the piece being dragged to disappear.
4. Lift the first finger. Note that the numerator of the fraction increases.
5. Decrease the numerator by one using the circular arrow buttons

**Comment by jbphet** [comment]
I added back the code that ignores a released piece if that piece has been removed from the model.  This what we've been referring to as "the workaround".  I met with @jonathanolson and @marlitas and discussed whether we could solve the problem further back in the input chain as we did for the reset-all case, but it would be a pretty big effort to do this, so we decided to leave the current fix in place.  There may be a time in the future when we work to add better handling for canceling pointers in multitouch scenarios.

All of the sims in the fractions suite have been passing on CT since this change was committed, so I'm closing this issue.
